artifact/rules.bzl

```diff
@@ -148,22 +148,43 @@ def artifact_file(name,
         **kwargs
     )
 
-script_template = """\
+script_template_gz = """\
 #!/bin/bash
 set -ex
 mkdir -p $BUILD_WORKSPACE_DIRECTORY/$1
-tar -xzf {artifact_location} -C $BUILD_WORKSPACE_DIRECTORY/$1 --strip-components={strip_components}
+tar -xzf {artifact_location} -C $BUILD_WORKSPACE_DIRECTORY/$1 --strip-components=2
+"""
+
+script_template_zip = """\
+#!/bin/bash
+set -ex
+mkdir -p $BUILD_WORKSPACE_DIRECTORY/$1
+tmp_dir=$(mktemp -d)
+unzip -qq {artifact_location} -d $tmp_dir
+mv -v $tmp_dir/{artifact_unpacked_name}/* $BUILD_WORKSPACE_DIRECTORY/$1/
+rm -rf {artifact_unpacked_name}
 """
 
 def _artifact_extractor_impl(ctx):
     artifact_file = ctx.file.artifact
+    artifact_extention = artifact_file.extension
+
+    if artifact_extention == 'zip': 
+        target_script_template = script_template_zip
+        artifact_unpacked_name = artifact_file.basename.replace('.zip', '')
+    elif artifact_extention == 'gz':
```

Comment:
I know it's annoying but we should probably do a full check for `.tar.gz` and alternative extensions that `tar` will support, such as `.tgz`, just to make sure the rule is going to be a bit more future proof! You may have to use some starlark string matching instead of the convenient `.extension` sadly.

Reply:
resolved in https://github.com/graknlabs/bazel-distribution/pull/253/commits/07cd24b22dad38db1a949adb58388f514abaa9e0